src/main/java/org/apache/commons/lang3/StringUtils.java

```diff
@@ -7193,6 +7193,45 @@ public static boolean isAllUpperCase(final CharSequence cs) {
         return true;
     }
 
+    /**
+     * <p>Checks if the CharSequence contains mixed casing of both uppercase and lowercase characters.</p>
+     *
+     * <p>{@code null} will return {@code false}.
+     * An empty String (length()=0) will return {@code false}.</p>
+     *
+     * <pre>
+     * StringUtils.isMixedCase(null)    = false
+     * StringUtils.isMixedCase("")      = false
+     * StringUtils.isMixedCase("aBc")   = true
+     * StringUtils.isMixedCase("ABC")   = false
+     * StringUtils.isMixedCase("abc")   = false
+     * StringUtils.isMixedCase("A c")   = false
+     * StringUtils.isMixedCase("A1c")   = false
+     * StringUtils.isMixedCase("a/C")   = false
+     * </pre>
+     *
+     * @param cs the CharSequence to check, may be null
+     * @return {@code true} if contains both uppercase and lowercase characters, and is non-null
+     */
+    public static boolean isMixedCase(final CharSequence cs) {
+        if (cs == null || isEmpty(cs)) {
+            return false;
+        }
+        boolean containsUppercase = false;
+        boolean containsLowercase = false;
+        final int sz = cs.length();
+        for (int i = 0; i < sz; i++) {
```

Comment:
Wouldn't it be more efficient to track the case of the first character then break and return false on the first occurrence of the other case? Saves iterating the rest of the character sequence if it has mixed case.